fluentui_drawer/src/main/java/com/microsoft/fluentui/tokenized/bottomsheet/BottomSheet.kt

```diff
@@ -0,0 +1,493 @@
+    expandable: Boolean,
+    peekHeight: Dp,
+    fullHeight: Float,
+    sheetHeight: Float?
+): Modifier {
+    val modifier = if (sheetHeight != null) {
+        val peekHeightPx = min(dpToPx(peekHeight), fullHeight * BottomSheetOpenFraction)
+        val anchors = if (!expandable) {
+            mapOf(
+                fullHeight to BottomSheetValue.Hidden,
+                fullHeight - min(sheetHeight, peekHeightPx) to BottomSheetValue.Shown
+            )
+        } else if (sheetHeight <= peekHeightPx) {
+            mapOf(
+                fullHeight to BottomSheetValue.Hidden,
+                fullHeight - sheetHeight to BottomSheetValue.Shown
+            )
+        } else {
+            mapOf(
+                fullHeight to BottomSheetValue.Hidden,
+                fullHeight - peekHeightPx to BottomSheetValue.Shown,
+                max(0f, fullHeight - sheetHeight) to BottomSheetValue.Expanded
+            )
+        }
+        if (sheetState.initialValue == BottomSheetValue.Expanded
+            && anchors.entries.firstOrNull { it.value == BottomSheetValue.Expanded } == null
+        ) {
+            throw IllegalArgumentException(
+                "BottomSheet initial value must not be set to Expanded " +
+                        "if the whole content is visible in Shown state itself"
+            )
+        }
+
+        Modifier.swipeable(
+            state = sheetState,
+            anchors = anchors,
+            orientation = Orientation.Vertical,
+            enabled = sheetState.currentValue != BottomSheetValue.Hidden,
+            resistance = null
+        )
+    } else {
+        Modifier
+    }
+
+    return this.then(modifier)
+}
+
+private fun Modifier.sheetHeight(expandable: Boolean, fullHeight: Float, peekHeight: Dp): Modifier {
+    val modifier = if (expandable) {
+        Modifier
+    } else {
+        Modifier.heightIn(
+            0.dp,
+            pxToDp(min(fullHeight * BottomSheetOpenFraction, dpToPx(peekHeight)))
+        )
+    }
+    return this.then(modifier)
+}
+
+private fun calculateFraction(a: Float, b: Float, pos: Float) =
+    ((pos - a) / (b - a)).coerceIn(0f, 1f)
+
+@Composable
+private fun Scrim(
+    color: Color,
+    onDismiss: () -> Unit,
+    fraction: () -> Float,
+    visible: Boolean
+) {
+    if (visible) {
+        val closeSheet = getString(Strings.CloseSheet)
+        val dismissModifier = if (visible) {
+            Modifier
+                .pointerInput(onDismiss) { detectTapGestures { onDismiss() } }
+                .semantics(mergeDescendants = true) {
+                    contentDescription = closeSheet
+                    onClick { onDismiss(); true }
+                }
+        } else {
+            Modifier
+        }
+
+        Canvas(
+            Modifier
+                .fillMaxSize()
+                .then(dismissModifier)
+                .testTag(BOTTOMSHEET_SCRIM_TAG)
+
+        ) {
+            drawRect(color = color, alpha = fraction())
+        }
+    }
+}
```

Comment:
Merge with Drawer's function if possible.

Reply:
Drawer & Bottomsheet focus requirement on scrim are different. Hence, cannot be merged.

Reply:
Can you explain the differences? Can we not make scrim similar for both?

Reply:
The only difference I see is semantics modifier which adds content description. Which function works for capturing focus here? Also calculateFraction function seems it can be combined. Possibly, when you create a common function, you can add a boolean param for requestFocus/contentDescription. Why I am saying this is, its possible we might need scrim for PopUp Window/menu as well.

Reply:
Created a Utils.kt & moved calculateFraction to it. Added TODO to revisit Scrim to have single implementation if possible.